### PR TITLE
[Ground Control] Platform improvements

### PR DESCRIPTION
**Pull request by davea**
This PR is a collection of small improvements to FMS that came about as part of the Ground Control site.

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Root.pm [hunk 1]
```diff
@@ -130,6 +142,27 @@ sub page_error : Private {
...
+    return if $c->user_exists || !FixMyStreet->config('LOGIN_REQUIRED');
+
+    my $whitelist = [
+        '^auth$',
+        '^auth/',
+        '^js/translation_strings\.[a-zA-Z]{2}(-[a-zA-Z]{2})?\.js',
+        '^[PACQM]{1}/', # various tokens that log the user in
+    ];
+
+    foreach (@$whitelist) {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Root.pm]: No reason for this to be an array reference, plus you could join the entries with `|` and do one regex? In fact, with `x` you could write it out as a multiline regex, think that's as clear, e.g.
```
my $whitelist = qr{
    ^auth(/|$)
  | ^js/translation_strings\.(.*?)\.js
  | ^[PACQM]/  # various tokens that log the user in
}x;
return if $c->request->path =~ $whitelist;
```

File: bin/make_css [hunk 2]
```diff
@@ -35,6 +35,8 @@ GetOptions(
     'style|t=s' => sub { die "Unknown style\n" unless $_[1] =~ /nested|expanded|compact|compressed/; $style = $_[1]; },
     'verbose' => \my $verbose,
     'watch' => \my $watch,
+    'script=s'=> \my $watch_script,
+    'script-param=s'=> \my $watch_script_param,
```
Comment (dracos) [inline bin/make_css]: If you put this in an array, you could pass multiple parameters (dunno if that's needed/wanted!)

File: perllib/FixMyStreet/App/Controller/Root.pm [hunk 3]
```diff
@@ -147,19 +147,33 @@ sub check_login_required : Private {
...
+        '^offline',
+    ];
+
     foreach (@$whitelist) {
         if ( $c->request->path =~ m#$_# ) {
             return;
         }
     }
 
+    foreach (@$blacklist) {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Root.pm]: Same as with the whitelist.

File: perllib/FixMyStreet/App/Controller/Root.pm [hunk 4]
```diff
@@ -130,6 +142,27 @@ sub page_error : Private {
...
+sub check_login_required : Private {
+    my ($self, $c) = @_;
+
+    return if $c->user_exists || !FixMyStreet->config('LOGIN_REQUIRED');
+
+    my $whitelist = [
+        '^auth$',
+        '^auth/',
+        '^js/translation_strings\.[a-zA-Z]{2}(-[a-zA-Z]{2})?\.js',
+        '^[PACQM]{1}/', # various tokens that log the user in
```
Comment (davea) [inline perllib/FixMyStreet/App/Controller/Root.pm]: Duh :)

**Comment by davea** [comment]
I think I've covered off all the points you raised, so this is ready for a re-review.

**Comment by dracos** [comment]
Nice :)

**Pull request merged**